src/txpasslib/test/test_context.py

```diff
@@ -0,0 +1,57 @@
+from hypothesis import strategies as s
+from hypothesis import given
+from passlib.context import CryptContext
+from testtools import TestCase
+from testtools.matchers import (
+    AfterPreprocessing, Equals, IsInstance, MatchesAll)
+from testtools.twistedsupport import succeeded
+from twisted._threads import createMemoryWorker
+
+from txpasslib.context import TxCryptContext
+from txpasslib.test.doubles import SynchronousReactorThreads
+from txpasslib.test.matchers import failed_with, performed
+
+
+class ContextTests(TestCase):
+    """
+    Tests for `txpasslib.context.CryptContext`.
+    """
+    @given(password=s.text())
+    def test_hash_and_verify(self, password):
+        """
+        Hashing a password fires with the hash; verifying the password against
+        that hash succeeds, while verifying a different password fails.
+        """
+        reactor = SynchronousReactorThreads()
+        worker, perform = createMemoryWorker()
+        ctx = TxCryptContext(
+            CryptContext(schemes=['plaintext']),
+            reactor, worker)
+        self.assertThat(
```

Comment:
So the error message is actually pretty great!  It might be nice to wrap the `ctx.verify` calls with named functions so a failed match would read like this:

```
testtools.matchers._impl.MismatchError: Differences: [
Differences: [
Differences: [
False != True
]: after <function unsuccessful_verify> on ''
]
]
```

But even so this actually makes me want to use testtools' matchers :)